pkg/project/registry/projectrequest/delegated/delegated_test.go

```diff
@@ -1,6 +1,80 @@
 package delegated
 
-import "testing"
+import (
+	"sync"
+	"testing"
+	"time"
 
-func TestDelegated(t *testing.T) {
+	kapi "k8s.io/kubernetes/pkg/api"
+
+	authorizationapi "github.com/openshift/origin/pkg/authorization/api"
+	authorizationclient "github.com/openshift/origin/pkg/authorization/client"
+	"github.com/openshift/origin/pkg/cmd/server/bootstrappolicy"
+)
+
+func TestDelegatedWait(t *testing.T) {
+	cache := &testReadOnlyPolicyBinding{}
+	storage := &REST{policyBindings: cache}
+
+	cache.binding = &authorizationapi.PolicyBinding{}
+	cache.binding.RoleBindings = map[string]*authorizationapi.RoleBinding{}
+	cache.binding.RoleBindings["anything"] = nil
+
+	callReturnedCh := testWait(storage)
+
+	select {
+	case <-callReturnedCh:
+		t.Errorf("too fast, should have blocked")
+	case <-time.After(1 * time.Second):
+	}
+
+	func() {
+		cache.lock.Lock()
+		defer cache.lock.Unlock()
+		cache.binding.RoleBindings[bootstrappolicy.AdminRoleName] = nil
```

Comment:
I have stopped it from happening by returning a copy of the rolebindings map in the fake list.
```go
func (t *testReadOnlyPolicyBinding) List(options *kapi.ListOptions) (*authorizationapi.PolicyBindingList, error) {
	t.lock.Lock()
	defer t.lock.Unlock()
	ret := &authorizationapi.PolicyBindingList{}
	returning := *t.binding
	copy := map[string]*authorizationapi.RoleBinding{}
	for k, v := range t.binding.RoleBindings {
		copy[k] = v
	}
	returning.RoleBindings = copy
	if t.binding != nil {
		ret.Items = []authorizationapi.PolicyBinding{returning}
	}

	return ret, nil
}
```
So my theory is that the lock protects from accessing the cache but not from accessing the underlying map which (as a reference) is being polled by the exponential loop.